src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import core.basesyntax.service.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.service.ShopServiceImpl;
+import core.basesyntax.service.date.DateConverter;
+import core.basesyntax.service.date.DateConverterImpl;
+import core.basesyntax.service.date.ReportGenerator;
+import core.basesyntax.service.date.ReportGeneratorImpl;
+import core.basesyntax.service.files.FileReaderCsv;
+import core.basesyntax.service.files.FileReaderCsvImpl;
+import core.basesyntax.service.files.FileWriterCsv;
+import core.basesyntax.service.files.FileWriterCsvImpl;
+import core.basesyntax.service.operation.BalanceOperationHandler;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.OperationStrategy;
+import core.basesyntax.service.operation.OperationStrategyImpl;
+import core.basesyntax.service.operation.PurchaseOperationHandler;
+import core.basesyntax.service.operation.ReturnOperationHandler;
+import core.basesyntax.service.operation.SupplyOperationHandler;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+        FileReaderCsv fileReader = new FileReaderCsvImpl();
+        List<String> inputReport = fileReader.read("fileToRead.csv");
+
+        Map<FruitTransaction.Operation, OperationHandler> operationHandler = new HashMap<>();
+        operationHandler.put(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler());
+        operationHandler.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperationHandler());
+        operationHandler.put(FruitTransaction.Operation.RETURN, new ReturnOperationHandler());
+        operationHandler.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
```

Comment:
Good job on utilizing the Strategy pattern to handle different operations. This way, adding new operations in the future will be a lot easier and won't require modifying existing code. [Avoid using switch-cases and if-else constructions. It is recommended to use the Strategy pattern instead.]